src/rust/ide/view/src/searcher/new.rs

```diff
@@ -0,0 +1,83 @@
+//! A stub module with new searcher GUI.
+
+use crate::prelude::*;
+
+
+// =============
+// === Entry ===
+// =============
+
+/// A structure describing a single entry in Searcher.
+#[allow(missing_docs)]
+#[derive(Clone,CloneRef,Debug,Default)]
+pub struct Entry {
+    pub label     : ImString,
+    pub is_folder : Immutable<bool>,
+    pub icon      : Icon,
+}
+
+/// The typewrapper for icon name.
+#[derive(Clone,CloneRef,Debug,Default)]
+pub struct Icon {
+    name : ImString
+}
+    
+// ===========
+// === FRP ===
+// ===========
+
+ensogl::define_endpoints! { <ID>
+    Input {
+        reset(),
+        directory_content (Vec<ID>,Entry),
```

Comment:
Please change it to `Rc<Vec<ID>>` (or make newtype `ImVec` for it pls). FRP clones these